### ISSUE TITLE
Extension Host Terminated unexpectedly

### ISSUE THREAD

**Issue by dipta007** [microsoft/vscode#104478]
- VSCode Version: 1.47.3
- OS Version: Ubuntu 18.04

Steps to Reproduce:

1. When I open a big project to work on, after some time I get this: 
`Extension Host Terminated unexpectedly`
2. And on the developer console log, there are a lot of messages like this one
```
[Extension Host] Error: EMFILE: too many open files, open '/home/......'
```

Does this issue occur when all extensions are disabled?: Yes/No
Yes

**Comment by shaft49** [comment]
Version: 1.48.0
OS Version: macOS 10.15.6

Removed all the extensions, installed fresh. Nothing works.
This happens even if I tried to open a single text file.

**Comment by alexdima** [comment]
@shaft49 Please open a separate issue.

@dipta007 Does the extension host crash when disabling all extensions? Press `F1` and then choose `Developer: Reload Window with Extensions Disabled`

![Kapture 2020-07-20 at 15 28 23](https://user-images.githubusercontent.com/5047891/87943040-c3b02a80-ca9d-11ea-847d-5f850fae8f05.gif)

**Comment by dipta007** [comment]
@alexdima yes I tried that too, and this error occurs only on a single project of mine. Not on the others.

**Comment by alexdima** [comment]
@dipta007 So it might be that you are running out of file descriptors only on this specific folder. Is this a very large folder with many subfolders and files? Have you searched online how to increase your file descriptors limit?

We have some steps on how to resolve ENOSPC here -- https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc 

I suppose there must be some steps to also resolve EMFILE by increasing some limit.

**Comment by dipta007** [comment]
@alexdima I have already follow the [guildeline](https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc) you provided. But the issue is not fixed yet.

**Comment by alexdima** [comment]
@dipta007 Have you tried `sysctl -w kern.maxfiles=20480`? (I found it online when searching for "EMFILE") - https://stackoverflow.com/q/8965606
